packages/backend-function/src/resolve_secret_banner.ts

```diff
@@ -7,31 +7,50 @@ import { SSM } from '@aws-sdk/client-ssm';
 export const internalAmplifyFunctionBannerResolveSecrets = async (
   client?: SSM
 ) => {
-  if (!client) {
-    client = new SSM();
-  }
-
-  const envPathObj: Record<string, string> = JSON.parse(
+  const envPathObject: Record<string, Record<string, string>> = JSON.parse(
     process.env.AMPLIFY_SECRET_PATHS ?? '{}'
   );
-  const paths = Object.values(envPathObj);
+  const paths = Object.keys(envPathObject);
 
   if (paths.length === 0) {
     return;
   }
 
-  const response = await client.getParameters({
-    Names: paths,
-    WithDecryption: true,
-  });
-  if (response.Parameters && response.Parameters?.length > 0) {
-    for (const parameter of response.Parameters) {
-      for (const [env, path] of Object.entries(envPathObj)) {
-        if (parameter.Name === path) {
-          process.env[env] = parameter.Value;
+  const resolveSecrets = async (paths: string[]) => {
+    if (!client) {
+      client = new SSM();
+    }
```

Comment:
I liked this better at the top of the main function

Reply:
Typescript was complaining that client is possibly `undefined` in `resolveSecrets` if that remained at the top of the main function

Reply:
Does it complain if we change the function signature to
```ts
export const internalAmplifyFunctionBannerResolveSecrets = async (client = new SSM())
```
?

Reply:
Yeah it does not complain with the function signature update so I have removed the if statement.